tests/rptest/tests/group_membership_test.py

```diff
@@ -113,3 +116,187 @@ def _check_no_duplicates():
+            rpk.produce(topic_1, key, payload, timeout=5)
+            rpk.produce(topic_2, key, payload, timeout=5)
+
+        group = "group"
+        rpk.consume(topic_1, group=group, n=100)
+        rpk.consume(topic_2, group=group, n=100)
+
+        metrics_offsets = self._get_offset_from_metrics(group)
+
+        self.redpanda.logger.debug(f"KEK: {metrics_offsets}")
+        sum_1 = 0
+        sum_2 = 0
+        assert len(metrics_offsets) == 6
+        for i in range(3):
+            assert (topic_1, str(i)) in metrics_offsets
+            assert metrics_offsets[(topic_1, str(i))] > 0
+            sum_1 += metrics_offsets[(topic_1, str(i))]
+
+            assert (topic_2, str(i)) in metrics_offsets
+            assert metrics_offsets[(topic_2, str(i))] > 0
+            sum_2 += metrics_offsets[(topic_2, str(i))]
+        assert sum_1 == 100
+        assert sum_2 == 100
+
+    @cluster(num_nodes=3)
+    def test_topic_recreation(self):
+        rpk = RpkTool(self.redpanda)
+        topic = "test_topic"
+        group = "g0"
+
+        def create_topic_and_check_metric():
+            spec = TopicSpec(name=topic, partition_count=1)
+            self.redpanda.create_topic(spec)
+            self._wait_topic_is_ready(topic)
+
+            for i in range(100):
+                payload = str(random.randint(0, 1000))
+                offset = rpk.produce(topic, "", payload, timeout=5)
+
+            metric_key = (topic, "0")
+            rpk.consume(topic, group=group, n=50)
+            metrics_offsets = self._get_offset_from_metrics(group)
+            assert metric_key in metrics_offsets
+            assert metrics_offsets[metric_key] == 50
+
+            self.redpanda.delete_topic(topic)
+
+        create_topic_and_check_metric()
+        metrics_offsets = self._get_offset_from_metrics(group)
+        assert metrics_offsets is None
+        create_topic_and_check_metric()
+
+    @cluster(num_nodes=3)
+    def test_leadership_transfer(self):
+        rpk = RpkTool(self.redpanda)
+        topic = "test_topic"
+        group = "g0"
+
+        spec = TopicSpec(name=topic)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic)
+
+        for i in range(100):
+            payload = str(random.randint(0, 1000))
+            offset = rpk.produce(topic, "", payload, timeout=5)
+
+        metric_key = (topic, "0")
+        rpk.consume(topic, group=group, n=50)
+        metrics_offsets = self._get_offset_from_metrics(group)
+        assert metric_key in metrics_offsets
+        assert metrics_offsets[metric_key] == 50
+        assert len(metrics_offsets) == 1
+
+        admin = Admin(redpanda=self.redpanda)
+
+        for n in self.redpanda.nodes:
+            admin.transfer_leadership_to(namespace="kafka_internal",
+                                         topic="group",
+                                         partition=0,
+                                         target=self.redpanda.idx(n))
```

Comment:
I think that this test should be a bit more detailed in its verification. Here is a rough sketch for what I have in mind:

```
loop N test iterations:

- choose a partition kafka_internal:group topic
- retrieve all related offset metrics from leader
- select a node that isn't the leader
- transfer leadership
- verify that eventually all the metrics are _not_ at the old node and are seen a the new node
```

it would make sense to have this test run with multiple active producer/consumers in the background. in this case you aren't really interested so much in the exactly value of the offset that is published but that the metrics themselves move around and expire properly.